src/cli/changesets/run.ts

```diff
@@ -0,0 +1,277 @@
+/* eslint-disable no-console */
+// Adapted from https://github.com/changesets/action/blob/21240c3cd1d2efa2672d64e0235a03cf139b83e6/src/run.ts
+import path from 'path';
+
+import type { Package } from '@manypkg/get-packages';
+import { getPackages } from '@manypkg/get-packages';
+import fs from 'fs-extra';
+import resolveFrom from 'resolve-from';
+
+import * as core from './coreAdapter';
+import * as gitUtils from './gitUtils';
+import * as github from './githubAdapter';
+import readChangesetState from './readChangesetState';
+import {
+  exec,
+  execWithOutput,
+  getChangedPackages,
+  getChangelogEntry,
+  getVersionsByDirectory,
+  sortTheThings,
+} from './utils';
+
+type CommonError = {
+  code: string;
+};
+
+const createRelease = async (
+  octokit: ReturnType<typeof github.getOctokit>,
+  {
+    pkg,
+    tagName,
+  }: {
+    pkg: Package;
+    tagName: string;
+  },
+) => {
+  try {
+    const changelogFileName = path.join(pkg.dir, 'CHANGELOG.md');
+
+    const changelog = await fs.readFile(changelogFileName, 'utf8');
+
+    const changelogEntry = getChangelogEntry(
+      changelog,
+      pkg.packageJson.version,
+    );
+    if (!changelogEntry) {
+      // we can find a changelog but not the entry for this version
+      // if this is true, something has probably gone wrong
+      throw new Error(
+        `Could not find changelog entry for ${pkg.packageJson.name}@${pkg.packageJson.version}`,
+      );
+    }
+
+    await octokit.repos.createRelease({
+      name: tagName,
+      tag_name: tagName,
+      body: changelogEntry.content,
+      prerelease: pkg.packageJson.version.includes('-'),
+      ...(await gitUtils.context(process.cwd())).repo,
+    });
+  } catch (err) {
+    // if we can't find a changelog, the user has probably disabled changelogs
+    if ((err as CommonError).code !== 'ENOENT') {
+      throw err;
+    }
+  }
+};
+
+type PublishOptions = {
+  script: string;
+  cwd?: string;
+};
+
+type PublishedPackage = { name: string; version: string };
+
+type PublishResult =
+  | {
+      published: true;
+      publishedPackages: PublishedPackage[];
+    }
+  | {
+      published: false;
+    };
+
+export async function runPublish({
+  script,
+  cwd = process.cwd(),
+}: PublishOptions): Promise<PublishResult> {
+  const octokit = github.getOctokit();
+  const [publishCommand, ...publishArgs] = script.split(/\s+/);
```

Comment:
What publish args do we support? Should we just hardcode the exec to `yarn changeset publish`?